pkg/network/ovn_kubernetes.go

```diff
@@ -415,35 +428,77 @@ func bootstrapOVNHyperShiftConfig(hc *HyperShiftConfig, kubeClient cnoclient.Cli
 	}
 
 	ovnHypershiftResult.ClusterID = hcp.Spec.ClusterID
-
-	route := &routev1.Route{}
-	gvr := schema.GroupVersionResource{
-		Group:    "route.openshift.io",
-		Version:  "v1",
-		Resource: "routes",
-	}
-	clusterClient := kubeClient.ClientFor(client.ManagementClusterName)
-	routeObj, err := clusterClient.Dynamic().Resource(gvr).Namespace(hc.Namespace).Get(context.TODO(), "ovnkube-sbdb", metav1.GetOptions{})
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			klog.Infof("Did not find ovnkube-sbdb route")
-		} else {
-			return nil, fmt.Errorf("Could not get ovnkube-sbdb route: %v", err)
+	for _, svc := range hcp.Spec.Services {
+		// TODO: instead of the hardcoded string use ServiceType hyperv1.OVNSbDb once the API is updated
+		if svc.Service == "OVNSbDb" {
+			ovnHypershiftResult.ServicePublishingStrategy = &svc.ServicePublishingStrategy
 		}
-	} else {
-		err := runtime.DefaultUnstructuredConverter.FromUnstructured(routeObj.UnstructuredContent(), route)
-		if err != nil {
-			return ovnHypershiftResult, err
+	}
+	if ovnHypershiftResult.ServicePublishingStrategy == nil {
+		klog.Warningf("service publishing strategy for OVN southbound database does not exist in hyperv1.HostedControlPlane %s/%s. Defaulting to route", hc.Name, hc.Namespace)
+		ovnHypershiftResult.ServicePublishingStrategy = &hyperv1.ServicePublishingStrategy{
+			Type: hyperv1.Route,
 		}
-		if route.Spec.Host == "" && route.Status.Ingress[0].Host == "" {
-			return ovnHypershiftResult, nil
+	}
+	switch ovnHypershiftResult.ServicePublishingStrategy.Type {
+	case hyperv1.Route:
+		{
+			route := &routev1.Route{}
+			gvr := schema.GroupVersionResource{
+				Group:    "route.openshift.io",
+				Version:  "v1",
+				Resource: "routes",
+			}
+			clusterClient := kubeClient.ClientFor(client.ManagementClusterName)
+			routeObj, err := clusterClient.Dynamic().Resource(gvr).Namespace(hc.Namespace).Get(context.TODO(), "ovnkube-sbdb", metav1.GetOptions{})
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					klog.Infof("Did not find ovnkube-sbdb route")
+				} else {
+					return nil, fmt.Errorf("could not get ovnkube-sbdb route: %v", err)
```

Comment:
nit: return `ovnHypershiftResult` and err?

Reply:
Went the other way and stopped returning `ovnHypershiftResult` on error as it is not used anyway.